DfE.Hexagonal.Model/School.cs

```diff
@@ -0,0 +1,38 @@
+﻿using DfE.Hexagonal.Common.Validation;
+
+namespace DfE.Hexagonal.Model
+{
+    public class School : IValidatable
+    {
+        public Guid Id { get; set; }
+        public string Name { get; set; }
+        public string Description { get; set; }
+        public DateTime OpenDate { get; set; }
+
+        public ValidationResult Validate()
+        {
+            bool isValid = true;
+            var errors = new Dictionary<string, string>();
+
+            if (Id.Equals(Guid.Empty))
```

Comment:
We explicitly guard against `null` strings below - should we guard against `null` here also?

Reply:
Are we talking about the Guid? (they aren't nullable)

Reply:
Interesting, thanks!

I was expecting setting the value to `null` to be a warning, but it appears to be a compilation error instead (due to `Guid` being a "value type" `struct` rather than a class/object, therefore cannot be assigned the value if not marked as nullable -- in contrast to non-value types which only show a warning and the program still compiles).

This might be blatantly obvious to folks, but it didn't register initially for me.

For anyone else following along/interested, I share my exploration on this below!

```csharp
#nullable enable
using System;

public class Program
{
    public static void Main()
    {
        var GuidExample = new GuidExample() {Id = null};
        var NullableGuidExample = new NullableGuidExample() {Id = null};
        var StringExample = new StringExample() {Id = null};
        var NullableStringExample = new NullableStringExample() {Id = null};
    }
}

public class GuidExample
{
    public Guid Id { get; set; }
}

public class NullableGuidExample
{
    public Guid? Id { get; set; }
}


public class StringExample
{
    public string Id { get; set; }
}

public class NullableStringExample
{
    public string? Id { get; set; }
}

```


... and here's a .NET Fiddle:

https://dotnetfiddle.net/bPadT4

![image](https://user-images.githubusercontent.com/96429508/207934699-2daf4cab-4b0e-4ee5-8c78-f3555327b961.png)